### ISSUE TITLE
ObjectUtils should lookup readResolve() with getMethod()

### ISSUE THREAD

**Issue by gregallen** [OpenHFT/Chronicle-Core#28]
This allows readResolve to be found as a default method, and allows this helper interface to be used with Chronicle Wire:

```
public interface MarshallableEnum<T extends Enum<T>> extends Marshallable
{
    default Object readResolve()
    {
        return Enum.valueOf((Class<T>) getClass(), name());
    }

    String name();
}
```

As things stand, the default readResolved is not used, and Marshallable Enums need to provide their own readResolve()

**Comment by gregallen** [comment]
Alternative implementation:

```
    default Object readResolve()
    {
        return  getClass().getEnumConstants()[ordinal()];
    }

    int ordinal();
```

**Comment by peter-lawrey** [comment]
I see your point in a general sense, however Enum are handled specially and
this shouldn't happen.

If you find an example where a new instance of an Enum is being created, it
is a bug and it would be very helpful if you could submit a unit test
demonstrating this.
ᐧ

**Comment by gregallen** [comment]
test covering 9 permutations :)

```
package greg;

import net.openhft.chronicle.bytes.Bytes;
import net.openhft.chronicle.wire.*;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.function.Function;
import java.util.function.Supplier;

import static org.hamcrest.Matchers.sameInstance;
import static org.junit.Assert.assertThat;

@RunWith(Parameterized.class)
public class WireTest {
    private Function<Bytes, Wire> createWire;

    @Parameterized.Parameters
    public static Iterable<Function<Bytes, Wire>> wires() {
        return Arrays.asList(b -> new TextWire(b), b -> new BinaryWire(b), b -> new RawWire(b));
    }

    public WireTest(Function<Bytes, Wire> createWire) {
        this.createWire = createWire;
    }

    @Test
    public void testEnumImplementingMarshallable() {
        assertThat(roundTrip(Person1::new).field, sameInstance(Marsh.MARSH));
    }

    @Test
    public void testEnumNotImplementingMarshallable() {
        assertThat(roundTrip(Person2::new).field, sameInstance(NoMarsh.NO_MARSH));
    }

    @Test
    public void testEnumImplementingMarshallableAndReadResolve() {
        assertThat(roundTrip(Person3::new).field, sameInstance(MarshAndResolve.MARSH_AND_RESOLVE));
    }

    <T extends Marshallable> T roundTrip(Supplier<T> supplier) {
        Wire wire = serialise(createWire, supplier.get());
        T deserialized = supplier.get();
        deserialized.readMarshallable(wire);
        return deserialized;
    }

    private static Wire serialise(Function<Bytes, Wire> createWire, Marshallable person) {
        Wire wire = createWire.apply(Bytes.elasticByteBuffer());
        person.writeMarshallable(wire);
        return wire;
    }

    private static void deserialize(Wire wire) {
        BytesTest.Person person = new BytesTest.Person();
        person.readMarshallable(wire);
    }

    enum Marsh implements Marshallable {
        MARSH
    }

    enum NoMarsh {
        NO_MARSH;
    }

    enum MarshAndResolve implements Marshallable {
        MARSH_AND_RESOLVE;

        MarshAndResolve readResolve() {
            return values()[ordinal()];
        }
    }

    static class Person1 extends AbstractMarshallable {
        private Marsh field = Marsh.MARSH;
    }

    static class Person2 extends AbstractMarshallable {
        private NoMarsh field = NoMarsh.NO_MARSH;
    }

    static class Person3 extends AbstractMarshallable {
        private MarshAndResolve field = MarshAndResolve.MARSH_AND_RESOLVE;
    }
}
```

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.20.102](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.20.102), [BOM-2.20.134](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.134)
